### PR TITLE
User register api and small refactoring

### PR DESCRIPTION
**Pull request by adeelnoor**

### CODE REVIEW

File: Controllers/UserController.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+            acount_status,
+            email,
+            profile_image_url,
+            phone_number,
+            user_selected_currency,
+            notification_token,
+          } = req.body;
+
+          // Validate required fields
+    if (!user_name) {
```
Comment (Mutahir2016) [inline Controllers/UserController.js:20]: here we have to add more validations

**Review by Mutahir2016 — APPROVED** [review]
Look good.

File: Controllers/UserController.js (hunk 1 above)
Comment (adeelnoor) [reply Controllers/UserController.js:20]: @Mutahir2016  yes please, add more validations and through separate errors. Make sure userName is optional but email, first and last name, country, city are mandatory field.

**Comment by adeelnoor** [comment]
Merging this PR atm @Mutahir2016 you can update validation while creatin apis

**Pull request merged**